ArrayOps.java

```diff
@@ -1,26 +1,173 @@
+        //int [] array={1,1,1,2,2};
+        //System.out.println(isSorted(array));
     }
     
+    //returns the missing integer in the array, between 0 and the array length
     public static int findMissingInt (int [] array) {
         // Write your code here:
-        return -1;
+        int sumArray=0;
+        int supposedSumArray=0;
+        int missing=0;
+
+
+        for(int i=1;i<=array.length;i++)
+        {
+            supposedSumArray+=i;
+        }
+
+        for(int j=0;j<array.length;j++)
+        {
+            sumArray+=array[j];
+        }
+
+        missing=supposedSumArray-sumArray;
+        return missing;
     }
 
+
     public static int secondMaxValue(int [] array) {
         // Write your code here:
-        return 0;
+        int max=array[0];
+        int tempI=0;
+        for(int i=1;i<array.length;i++)
+        {
+            if(array[i]>max)
+            {
+                max=array[i];
+                tempI=i;
+            }
+        }
+
+        int max1=array[0];
+        for(int j=1;j<array.length;j++)
+        {
+            
+            if(array[j]>max1 && tempI!=j) //&& array[j]!=max
+            {
+                max1=array[j];
+            }
+            else
+            {
+                if(array[j]==max && j!=tempI)
+                {
+                    max1=array[j];
+                    j=array.length; //break;
+                }
+            }
+        }
+        return max1;
     }
 
     public static boolean containsTheSameElements(int [] array1,int [] array2) {
         // Write your code here:
-        return false;
+        boolean sameElements1=true;
+        boolean sameElements2=true;
+        
+            for(int i=0; i<array1.length; i++)
+            {
+                for(int j=0; j<array2.length; j++)
+                {
+                    if(array1[i]==array2[j])
+                    {
+                        sameElements1=true;
+                        j=array2.length;
+                    }
+                    else
+                    {
+                        sameElements1=false;
+                    }
+                }
+            
+                if(sameElements1==false)
+                {
+                    break;
+                }
+            }
+        
+        
+            for(int i=0; i<array2.length; i++)
+            {
+                for(int j=0; j<array1.length; j++)
+                {
+                    if(array2[i]==array1[j])
+                    {
+                        sameElements2=true;
+                        j=array1.length;
+                    }
+                    else
+                    {
+                        sameElements2=false;
+                    }
+                }
+            
+                if(sameElements2==false)
+                {
+                    break;
+                }
+            }
+        
+
+        
+        return sameElements1&&sameElements2;
     }
```

Comment:
Seems like you implemented the same logic twice and it cause code duplication.
In that cause you should create a function and use it twice instead.
```suggestion
        return isSubsetOf(array1, array2) && isSubsetOf(array2, array1);
    }
    
    public static boolean isSubsetOf(int[] array1, int[] array2) {
            boolean sameElements1=true;   
            for(int i=0; i<array1.length; i++)
            {
                for(int j=0; j<array2.length; j++)
                {
                    if(array1[i]==array2[j])
                    {
                        sameElements1=true;
                        j=array2.length;
                    }
                    else
                    {
                        sameElements1=false;
                    }
                }
            
                if(sameElements1==false)
                {
                    break;
                }
            }
        return sameElements1;
    }
```